ocp_utilities/data_collector.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import os
+
+from ocp_resources.datavolume import DataVolume
+from ocp_resources.namespace import Namespace
+from ocp_resources.pod import Pod
+from ocp_resources.project import ProjectRequest
+from ocp_resources.resource import NamespacedResource, get_client
+from ocp_resources.virtual_machine import VirtualMachine
+from pytest_testconfig import config as py_config
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def collect_vmi_data(vmi, directory=None):
+    write_to_file(
+        file_name=f"vmi-{vmi.name}.yaml",
+        content=vmi.instance.to_str(),
+        base_directory=directory,
+    )
+
+    virt_launcher_pod = vmi.virt_launcher_pod
+    write_to_file(
+        file_name=f"{virt_launcher_pod.name}.log",
+        content=virt_launcher_pod.log(container="compute"),
+        base_directory=directory,
+    )
+    write_to_file(
+        file_name=f"{virt_launcher_pod.name}.yaml",
+        content=virt_launcher_pod.instance.to_str(),
+        base_directory=directory,
+    )
+
+
+def collect_data_volume_data(dyn_client, directory=None):
+    cdi_pod_prefixes = ("importer", "cdi-upload")
+    for pod in Pod.get(dyn_client=dyn_client):
+        pod_name = pod.name
+        if pod_name.startswith(cdi_pod_prefixes) or pod_name.endswith("source-pod"):
+            write_to_file(
+                file_name=f"{pod_name}.log", content=pod.log(), base_directory=directory
+            )
+            write_to_file(
+                file_name=f"{pod_name}.yaml",
+                content=pod.instance.to_str(),
+                base_directory=directory,
+            )
+
+
+def collect_data(directory, resource_object, collect_pod_logs):
+    LOGGER.info(f"Collecting instance data for {resource_object.name}")
+
+    if resource_object.kind == Pod.kind:
+        collect_pods_data(
+            pods_list=[resource_object],
+            collect_pod_logs=collect_pod_logs,
+            base_directory=directory,
+        )
+        return
+
+    if resource_object.kind == ProjectRequest.kind:
+        resource_object = Namespace(name=resource_object.name)
+
+    write_to_file(
+        file_name=f"{resource_object.name}.yaml",
+        content=resource_object.instance.to_str(),
+        base_directory=directory,
+    )
+    if resource_object.kind == VirtualMachine.kind and resource_object.vmi:
+        collect_vmi_data(vmi=resource_object.vmi, directory=directory)
+
+    if resource_object.kind == DataVolume.kind:
+        collect_data_volume_data(dyn_client=resource_object.client, directory=directory)
+
+
+def prepare_test_data_dir(item, prefix, logs_path=None):
+    test_cls_name = item.cls.__name__ if item.cls else ""
```

Comment:
`test_cls_name`, are we bounded to tests? maybe rename it?

Reply:
as discussed, this will remain a pytest item function